### PR TITLE
Improve documentation on structured logging

### PR DESCRIPTION
**Pull request by vy**

### CODE REVIEW

File: src/site/antora/modules/ROOT/pages/manual/layouts.adoc [hunk 1]
```diff
@@ -29,10 +29,20 @@ This page will try to answer following questions:
 
 This section introduces you to some common concerns shared by almost all <<collection,predefined layouts>> that you need to be aware of while using them.
 
+[#structured-logging]
+=== Structured logging
+
+Log4j strives to provide top of the class support for *structured logging*.
+To create an end-to-end experience, it provides several xref:manual/messages.adoc#collection-structured[structured message types] along with layouts supporting structured logging.
+
+include::partial$manual/structured-logging.adoc[]
```
Comment (ppkarwasz) [inline src/site/antora/modules/ROOT/pages/manual/layouts.adoc:38]: I think we don't need to collapse the "Structured logging" definition here.

Let us start with the include and then continue with "Log4j strives to provide..."

**Review by ppkarwasz — CHANGES_REQUESTED** [review]
Nice job.

I added some notes below.

File: src/site/antora/modules/ROOT/pages/manual/eventlogging.adoc [hunk 2]
```diff
@@ -17,236 +17,35 @@
...
-import java.io.IOException;
-
-public class UserActionServlet extends HttpServlet {
-    
-    protected void doPost(HttpServletRequest request, HttpServletResponse response)
-            throws ServletException, IOException {
-        String userId = request.getParameter("userId");
-        String action = request.getParameter("action");
-        String details = request.getParameter("details");
+private static final Marker EVENT_MARKER = MarkerManager.getMarker("EVENT");
```
[the remark this answers is not in the document]
Comment (vy) [reply src/site/antora/modules/ROOT/pages/manual/eventlogging.adoc:46]: Let's not create more use cases for a class we are deprecating.

File: src/site/antora/modules/ROOT/pages/manual/eventlogging.adoc [hunk 3]
```diff
@@ -17,236 +17,35 @@
 
 = Event Logger
 
-The `EventLogger` class provides a mechanism for logging significant events 
-in an application using structured data. This approach is beneficial for 
-tracking user actions, monitoring system behavior, and debugging issues.
+link:../javadoc/log4j-api/org/apache/logging/log4j/EventLogger.html[`EventLogger`] is a convenience to log xref:manual/messages.adoc#StructuredDataMessage[`StructuredDataMessage`]s, which format their content in a way compliant with https://datatracker.ietf.org/doc/html/rfc5424#section-6[the Syslog message format described in RFC 5424].
```
[the remark this answers is not in the document]
Comment (vy) [reply src/site/antora/modules/ROOT/pages/manual/eventlogging.adoc:20]: Fixed in 9b576968103f5f25acfd530e09ac5218cb92a4c8.

File: src/site/antora/modules/ROOT/pages/manual/layouts.adoc (hunk 1 above)
Comment (vy) [reply src/site/antora/modules/ROOT/pages/manual/layouts.adoc]: Fixed in e6f2cd98126de18f4db9e6e8b75439dca222f27f.

File: src/site/antora/modules/ROOT/pages/manual/messages.adoc [hunk 4]
```diff
@@ -217,7 +266,7 @@ In such a scenario, if you also have custom message types, consider implementing
 [#extending]
 == Extending
 
-If <<collection[predefined message types] fall short of addressing your needs, you can extend from the link:../javadoc/log4j-api/org/apache/logging/log4j/message/Message.html,`Message`>> interface to either create your own message types or make your domain models take control of the message formatting.
+If <<collection,predefined message types>> fall short of addressing your needs, you can extend from the link:../javadoc/log4j-api/org/apache/logging/log4j/message/Message.html[`Message`] interface to either create your own message types or make your domain models take control of the message formatting.
```
[the remark this answers is not in the document]
Comment (vy) [reply src/site/antora/modules/ROOT/pages/manual/messages.adoc:273]: Fixed in a8090af864c0fa8c9e8c2a983c8fa10f2e3dfc2e.

**Pull request merged**